src/spikeinterface/core/baserecording.py

```diff
@@ -233,8 +233,14 @@ def get_duration(self, segment_index=None) -> float:
             The duration in seconds
         """
         segment_index = self._check_segment_index(segment_index)
-        segment_num_samples = self.get_num_samples(segment_index=segment_index)
-        segment_duration = segment_num_samples / self.get_sampling_frequency()
+
+        if self.has_time_vector(segment_index):
+            times = self.get_times(segment_index)
+            segment_duration = times[-1] - times[0] + (1 / self.get_sampling_frequency())
```

Comment:
Got you, makes sense. Also, if you were to have only one sample this will give you a segment_duration of 0 which does not make sense while your formula gives you 1/ sampling_rate which is correct.

That said, it might be that something set times because the sampling rate is not quite right or something else and while that might be a good approximation it would be better if you did:  `duration_of_samples = np.diff(times).mean()` 

It does not look very costly:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/6cebfa1b-31af-4389-b0e2-de8e1316c23b)

Reply:
Nice! I wasn't sure how to handle this, thanks for the profiling also! Will add

Reply:
Wait, I think that's wrong. Maybe not:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/182494b0-6f50-4716-980f-0bca0a5cc2ff)

Reply:
I see yes, on my crappy laptop I can't allocate the space to create the 24h recording 😆 In this case, because the `sampling_rate` should be a good approximation to what the true sampling rate will be (I think?), and it is only adding one sample step value at the end, maybe we stick with the sampling rate? WDYT?

Reply:
Yes, my hunch is to go for the performant code (adding the 1/sr) giving that is a small adjustment.